server/src/main/java/org/opensearch/index/translog/transfer/BaseTranslogTransferManager.java

```diff
@@ -236,22 +271,95 @@ public boolean downloadTranslog(String primaryTerm, String generation, Path loca
         String translogFilename = Translog.getFilename(Long.parseLong(generation));
-        downloadToFS(translogFilename, location, primaryTerm);
+        downloadTranslogFilesToFS(translogFilename, location, primaryTerm, generation);
         return true;
     }
 
-    private void downloadToFS(String fileName, Path location, String primaryTerm) throws IOException {
+    /**
+     * Retrieves and processes translog files from a remote store.
+     *
+     * <p>This method first attempts to download the "translog.tlog" file from the remote store
+     * and copy it to the FS. If the downloaded file does not contain specific
+     * metadata, it then tries to download the "translog.ckp" file from the remote store and
+     * copy it to the local FS. If the "translog.tlog" file contains the required
+     * metadata, its content is used to write the "translog.ckp" file to FS.
+     *
+     * @param fileName     The name of the translog file (e.g., "translog.tlog").
+     * @param location     The local file system path where the translog files will be stored.
+     * @param primaryTerm  The primary term associated with the translog files.
+     * @param generation   The generation associated with the translog files.
+     * @throws IOException If an I/O error occurs during the file operations.
+     */
+    private void downloadTranslogFilesToFS(String fileName, Path location, String primaryTerm, String generation) throws IOException {
         Path filePath = location.resolve(fileName);
         // Here, we always override the existing file if present.
         // We need to change this logic when we introduce incremental download
-        if (Files.exists(filePath)) {
-            Files.delete(filePath);
+        deleteFileIfExists(filePath);
+
+        boolean downloadStatus = false;
+        long bytesToRead = 0, downloadStartTime = System.nanoTime();
+        Map<String, String> metadata;
+
+        try (
+            InputStreamWithMetadata inputStreamWithMetadata = transferService.downloadBlobWithMetadata(
+                remoteDataTransferPath.add(primaryTerm),
+                fileName
+            )
+        ) {
+            InputStream inputStream = inputStreamWithMetadata.getInputStream();
+            metadata = inputStreamWithMetadata.getMetadata();
+
+            bytesToRead = inputStream.available();
+            Files.copy(inputStream, filePath);
+            downloadStatus = true;
+
+        } finally {
+            remoteTranslogTransferTracker.addDownloadTimeInMillis((System.nanoTime() - downloadStartTime) / 1_000_000L);
+            if (downloadStatus) {
+                remoteTranslogTransferTracker.addDownloadBytesSucceeded(bytesToRead);
+            }
+        }
+
+        // Mark in FileTransferTracker so that the same files are not uploaded at the time of translog sync
+        fileTransferTracker.add(fileName, true);
+
+        try {
+            if (!isMetadataContainsCheckpointData(metadata)) {
```

Comment:
nit - should we first have if (isMetadataContainsCheckpointData(metadata)), intuitive to understand.